test/e2e/framework/pod/create.go

```diff
@@ -247,13 +250,109 @@ func MakeSecPod(podConfig *Config) (*v1.Pod, error) {
 		volumeIndex++
 	}
 
-	podSpec.Spec.Containers[0].VolumeMounts = volumeMounts
-	podSpec.Spec.Containers[0].VolumeDevices = volumeDevices
-	podSpec.Spec.Volumes = volumes
+	podSpec.Containers[0].VolumeMounts = volumeMounts
+	podSpec.Containers[0].VolumeDevices = volumeDevices
+	podSpec.Volumes = volumes
 	if runtime.GOOS != "windows" {
-		podSpec.Spec.SecurityContext.SELinuxOptions = podConfig.SeLinuxLabel
+		podSpec.SecurityContext.SELinuxOptions = podConfig.SeLinuxLabel
 	}
 
-	SetNodeSelection(&podSpec.Spec, podConfig.NodeSelection)
-	return podSpec, nil
+	SetNodeSelection(podSpec, podConfig.NodeSelection)
+	return podSpec
+}
+
+// NodeOSDistroIs returns true if the distro is the same as `--node-os-distro`
+// the package framework/pod can't import the framework package (see #81245)
+// we need to check if the --node-os-distro=windows is set and the framework package
+// is the one that's parsing the flags, as a workaround this method is looking for the same flag again
+// TODO: replace with `framework.NodeOSDistroIs` when #81245 is complete
+func NodeOSDistroIs(distro string) bool {
+	var nodeOsDistro *flag.Flag = flag.Lookup("node-os-distro")
+	if nodeOsDistro != nil && nodeOsDistro.Value.String() == distro {
+		return true
+	}
+	return false
+}
+
+// GenerateScriptCmd generates the corresponding command lines to execute a command.
+// Depending on the Node OS is Windows or linux, the command will use powershell or /bin/sh
+func GenerateScriptCmd(command string) []string {
```

Comment:
I think current windows test image Agnhost can support bash script. So we no longer need to convert to powershell at least in most cases.

Reply:
this is good to know, do you think we should make that change in this PR as well?

Reply:
can work on a different PR. We can remove it after testing all related tests and remove it together if possible.